romeo/src/stacks_client.rs

```diff
@@ -180,15 +180,30 @@ impl StacksClient {
             .unwrap()
     }
 
-    fn nonce_url(&self) -> reqwest::Url {
+    fn cachebust(&self, mut url: reqwest::Url) -> reqwest::Url {
         let mut rng = thread_rng();
         let random_string: String = (0..16).map(|_| rng.sample(Alphanumeric) as char).collect();
 
-        // We need to make sure node returns the uncached nonce, so we add a cachebuster
+        let mut query = url
+            .query()
+            .map(|query| query.to_string())
+            .unwrap_or_default();
+
+        query = if query.is_empty() {
+            format!("cachebuster={}", random_string)
+        } else {
+            format!("{}&cachebuster={}", query, random_string)
+        };
+
+        url.set_query(Some(&query));
+
+        url
+    }
+
+    fn nonce_url(&self) -> reqwest::Url {
         let path = format!(
-            "/extended/v1/address/{}/nonces?cachebuster={}",
+            "/extended/v1/address/{}/nonces",
```

Comment:
Thoughts on putting this into a constant?

Reply:
I usually put stuff in constants when they're used more than once. I can't imagine this being used anywhere else.